src/main/java/com/onlinebookstore/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,148 @@
+import com.onlinebookstore.onlinebookstore.model.orders.OrderStatus;
+import com.onlinebookstore.onlinebookstore.repository.OrderRepository;
+import com.onlinebookstore.onlinebookstore.repository.ShoppingCartRepository;
+import com.onlinebookstore.onlinebookstore.repository.UserRepository;
+import com.onlinebookstore.onlinebookstore.service.interfaces.OrderService;
+import com.onlinebookstore.onlinebookstore.service.interfaces.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    private final OrderRepository orderRepository;
+    private final UserRepository userRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    public OrderResponseDto placeOrder(OrderRequestDto orderRequestDto) {
+        User user = userRepository.findById(orderRequestDto.getUserId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "User not found with id: " + orderRequestDto.getUserId()));
+
+        ShoppingCart cart = shoppingCartRepository.findByUserId(user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Shopping cart not found for user id: " + user.getId()));
+
+        Order order = orderMapper.toModel(orderRequestDto);
+        order.setUser(user);
+        order.setOrderStatus(OrderStatus.PENDING);
+        order.setOrderItems(cart.getCartItems().stream()
+                .map(cartItem -> {
+                    OrderItem orderItem = new OrderItem();
+                    orderItem.setOrder(order);
+                    orderItem.setBook(cartItem.getBook());
+                    orderItem.setQuantity(cartItem.getQuantity());
+                    orderItem.setPrice(cartItem.getBook().getPrice());
+                    return orderItem;
+                }).collect(Collectors.toSet()));
+        order.setTotal(cart.getCartItems().stream()
+                .map(cartItem -> cartItem.getBook()
+                        .getPrice()
+                        .multiply(BigDecimal
+                                .valueOf(cartItem
+                                        .getQuantity())))
+                .reduce(BigDecimal.ZERO, BigDecimal::add));
+        order.setOrderDate(LocalDateTime.now());
+        order.setShippingAddress(orderRequestDto.getShippingAddress());
+
+        Order savedOrder = orderRepository.save(order);
+
+        cart.getCartItems().forEach(cartItem -> shoppingCartService
+                .removeBookFromCart(cartItem.getId(), user.getId()));

```

Comment:
You can delete all cartItems using one DB request, not reason to make N requests